content/articles/build-a-static-nextjs-markdown-files-blog-with-typescript-mdx-tailwindcss/index.md

```diff
@@ -0,0 +1,660 @@
+    return {
+        props: {
+            source: mdxSource,
+            frontMatter: data
+        }
+    }
+}
+
+export const getStaticPaths: GetStaticPaths = () => {
+    const posts = getAllPosts(['slug']); //only get the slug from posts 
+
+    const paths = posts.map((post) => ({ // map through to return post paths
+        params: {
+            slug: post.slug
+        }
+    }));
+
+    return {
+        paths,
+        fallback: false
+    }
+}
+```
+
+This dynamic file allows you to set your server-side and client-side as follows;
+
+On the server-side;
+
+- Fetch the content of the current post using `getStaticProps()`. The data of the article is serialized and returned as `source`, and `frontMatter`.
+- Fetch the paths of the post at build time from the `getStaticPaths()`. Also, return `fallback` to `false` so that every post path that is not generated at build time will generate a `404` error.
+
+On the client-side;
+
+- Get the `source`, and the `frontMatter` sent from the server.
+- Set the data to the application context using the consumer hook.
+- Check whether the page is building and returning a loading text.
+- Display the article content. The `source` and the `components` as shown in the `MDXRemote` component.
+
+Ensure that the development server is still running, and test if this works. Click on any article from the home page, and a single article page should be loaded.
+
+![specific-article-page](/engineering-education/build-a-static-nextjs-markdown-files-blog-with-typescript-mdx-tailwindcss/specific-article-page.PNG)
+
+### Deploy to Vercel
+To deploy to Vercel, ensure you push/publish your code to a GitHub repository first. [Login](https://vercel.com/login) to your Vercel dashboard or [register](https://vercel.com/signup) if you don't have one.
+
+From your [dashboard](https://vercel.com/dashboard), click on [New Project](https://vercel.com/new). Ensure that you have logged in to your GitHub account, select it as your Git provider, and then search and import your project.
+
+Enter your preferred project name and then click Deploy
+
+![vercel-deployment-conf](/engineering-education/build-a-static-nextjs-markdown-files-blog-with-typescript-mdx-tailwindcss/vercel-deployment-conf.PNG)
+
+After the deployment is done, click on the generated preview, and you will be redirected to your hosted blog application which you can share with friends and the general community.
+
+![hosted-blog-application](/engineering-education/build-a-static-nextjs-markdown-files-blog-with-typescript-mdx-tailwindcss/hosted-blog-application.PNG)
+
+### Conclusion
+Next.js is an amazing React-based framework. It allows you to work with almost any aspect of bot server-side and client-side content. It is very lightweight and allows you to create full fledge fast applications. In this tutorial, we built a blog application with Next.js, TypeScript, MDX, and Tailwind CSS and deployed it to Vercel. I hope you found the whole stack worth learning.
```

Comment:
It would be great if you could add some reference links that the reader could go through